### ISSUE TITLE
Can't push pull request branch

### ISSUE THREAD

**Issue by aweris** [peter-evans/create-pull-request#633]
### Subject of the issue

Getting error `Pushing pull request branch to 'origin/promote'` after upgrade v3

```
 /usr/bin/git push --force-with-lease origin HEAD:refs/heads/promote
  To https://github.com/<retacted>
   ! [rejected]          HEAD -> promote (stale info)
  error: failed to push some refs to 'https://github.com/<retacted>'
  Error: The process '/usr/bin/git' failed with exit code 1
```

### Steps to reproduce

Not sure if it's reproducible outside of our repo. We're using the `on schedule` action to create image promotion PR's. We only upgraded v3 because of the `The set-env command is disabled.` warnings.

Possibly, create a new branch not exist remotely could reproduces the error.

**Comment by fglaeser** [comment]
Hi @aweris , I have the same problem, after I merged the first PR and delete the branch, the action starts to showing that same message in every run.

**Comment by peter-evans** [comment]
Hi @aweris 

From the error message you provided it appears to be rejecting the push because of the use of `--force-with-lease`. You can see it says `stale info`. My understanding is that this would happen when changes have been made to the remote branch in-between the action fetching the branch and then pushing. Do you have other actions, workflows, or even users that are making commits to the `origin/promote` branch?

It would be helpful to see your entire workflow, just in case the cause is something about the way your workflow is written.

**Comment by aweris** [comment]
Hmm, just I recreated the branches and disabled the auto branch delete. It's working now. it's possible related to the difference between upstream and local. It's possible to solve the issue with `push --set-upstream origin`

**Comment by aweris** [comment]
@peter-evans, we're reusing the branch name for scheduled events for able to keep the system consistent. 

After merging the PR, branch, `origin/promote` deleted and recreated with the next scheduled action.

**Comment by peter-evans** [comment]
> It's possible to solve the issue with `push --set-upstream origin`

Where are you running this? In the workflow?
Please show me your workflow. I want to understand what is happening here and try to reproduce it.

**Comment by aweris** [comment]
I'll try to fetch all git history with new PR. I think it's worth to try

Workflow: 

```
name: scheduled release

on:
  schedule:
    # The shortest interval you can run scheduled workflows is once every 5 minutes.
    - cron:  '*/5 * * * *'
      
jobs:
  check-release:
    name: check

    runs-on: self-hosted

    steps:
      - name: Checkout
        uses: actions/checkout@v2

      - name: Check
        id: check
        run: <retacted>

      - name: Sync
        if: steps.check.outputs.triggered == 'true'       
        run: <retacted>

      - name: Create Pull Request
        if: steps.check.outputs.triggered == 'true'
        uses: peter-evans/create-pull-request@v3
        with:
          commit-message: <retacted>
          title: <retacted>
          body: <retacted>
          branch: promote
          reviewers: <retacted>
          labels: |
            automated
            promote
```

**Comment by aweris** [comment]
@peter-evans Just able to reproduce the error.

```
fatal: The current branch promote has no upstream branch.
To push the current branch and set the remote as upstream, use

    git push --set-upstream origin promote
```

and the current push is

```
 /usr/bin/git push --force-with-lease origin HEAD:refs/heads/promote
```

**Comment by sarathnairck12** [comment]
@peter-evans I am also seeing similar issue

```
Run peter-evans/create-pull-request@v2
Error: Unable to process command '::set-env name=pythonLocation::/opt/hostedtoolcache/Python/3.9.0/x64' successfully.
Error: The `set-env` command is disabled. Please upgrade to using Environment Files or opt into unsecure command execution by setting the `ACTIONS_ALLOW_UNSECURE_COMMANDS` environment variable to `true`. For more information see: https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/
Error: Unable to process command '::add-path::/opt/hostedtoolcache/Python/3.9.0/x64' successfully.
Error: The `add-path` command is disabled. Please upgrade to using Environment Files or opt into unsecure command execution by setting the `ACTIONS_ALLOW_UNSECURE_COMMANDS` environment variable to `true`. For more information see: https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/
Error: Unable to process command '::add-path::/opt/hostedtoolcache/Python/3.9.0/x64/bin' successfully.
Error: The `add-path` command is disabled. Please upgrade to using Environment Files or opt into unsecure command execution by setting the `ACTIONS_ALLOW_UNSECURE_COMMANDS` environment variable to `true`. For more information see: https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/
/opt/hostedtoolcache/Python/3.9.0/x64/bin/pip install --requirement /home/runner/work/_actions/peter-evans/create-pull-request/v2/dist/cpr/requirements.txt --no-index --find-links=/home/runner/work/_actions/peter-evans/create-pull-request/v2/dist/vendor
```

**Comment by aweris** [comment]
@sarathnairck12, 

This error is related to a recent security update in runners. The command basically disabled. You can try to upgrade `v3` since the new version is written in Typescript. It's not using `set-env` anymore.

Blog: https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/

**Comment by peter-evans** [comment]
@sarathnairck12 As mentioned by @aweris, please update to `v3` of the action to fix your issue. See [here](https://github.com/peter-evans/create-pull-request/blob/master/docs/updating.md) for notes about differences between the versions.

**Comment by peter-evans** [comment]
> Just able to reproduce the error.

@aweris Please could you explain in detail how you are reproducing the error so I can attempt to reproduce it, too.

Also, I'm very confused about a number of things in this thread. Please answer these questions:

> disabled the auto branch delete

What feature are you referring to here? Is it the `delete-branch` input on the action?

> It's possible to solve the issue with `push --set-upstream origin`

Did you run this to fix the issue? Where did you run it?

> I'll try to fetch all git history with new PR. I think it's worth to try

What does this mean? Are you referring to the way you were checking out the repository in the workflow?
The workflow you shared with me just uses `actions/checkout@v2` with default settings. Is the workflow you shared with me not the original workflow that experienced the error?

@fglaeser Since you also experienced this error, perhaps you could share your workflow and any detail that would help me reproduce this. Are you also running on `self-hosted` by any chance? If so, it may be related to the state of the local repository on the machine executing the self-hosted runner.

**Comment by aweris** [comment]
> @aweris Please could you explain in detail how you are reproducing the error so I can attempt to reproduce it, too.

Since this workflow working on the schedule. We're getting this error periodically. Just connected the current runner and make changes manually and try to push using commands printed in the workflow log. 

I think the main difference here is `self-hosted` runners. 

> What feature are you referring to here? Is it the delete-branch input on the action?

I'm talking about `Automatically delete head branches` for the repository. After creating the `promote` branch remotely. I'm able to create new PR from CI.

> Did you run this to fix the issue? Where did you run it?

It did not fix it. Used `actions/checkout@v2` , `fetch-depth: 0` for it. I just suspected missing git history; it seems unrelated to the problem.

The workflow I shared is what I use originally. This workflow is part of our deployment process. After the upgrade, it stopped working, so currently trying to find a workaround.

**Comment by fglaeser** [comment]
Hi @peter-evans, our problem started after merged the first pull request and delete the branch after that. To solve the problem I've changed the branch name for the PR and is working, the only thing different is now we don't delete the branch after merge. I will try to reproduce the error deleting the branch after merge to confirm.

And Yes, we are using self-hosted.

BR

**Comment by peter-evans** [comment]
I've managed to reproduce the issue. It does seem to be a problem with running the action on a self-hosted runner. Not deleting the branch after merge seems to be a temporary workaround. I'm now investigating what the cause is.

**Comment by sarathnairck12** [comment]
> @sarathnairck12 As mentioned by @aweris, please update to `v3` of the action to fix your issue. See [here](https://github.com/peter-evans/create-pull-request/blob/master/docs/updating.md) for notes about differences between the versions.

Thanks @peter-evans This fixed my issue

**Comment by peter-evans** [comment]
@aweris @fglaeser I have a workaround until I can fix the action. Add the following line to your workflow after the checkout step.

```diff
    steps:
      - uses: actions/checkout@v2

+     - run: git remote prune origin

      # Make changes to pull request here

      - name: Create Pull Request
        uses: peter-evans/create-pull-request@v3
```

**Comment by peter-evans** [comment]
This should now be fixed in the latest release (`v3`/`v3.5.1`)
